stix/idl/processing/imaging/stx_imaging_pipeline.pro

```diff
@@ -104,14 +145,39 @@ function stx_imaging_pipeline, stix_uid, time_range, energy_range, bkg_uid=bkg_u
   mapcenter = stx_hpc2stx_coord(xy_flare, aux_data)
   flare_loc  = mapcenter
 
-  ; Compute calibrated visibilities
-  if keyword_set(no_small) then $
-     vis = stx_construct_calibrated_visibility(path_sci_file, time_range, energy_range, mapcenter, $
-                                               path_bkg_file=path_bkg_file, xy_flare=flare_loc, /no_small, sumcase='TOP+BOT') $
-     else vis = stx_construct_calibrated_visibility(path_sci_file, time_range, energy_range, mapcenter, $
-                                               path_bkg_file=path_bkg_file, xy_flare=flare_loc)
+  ; Next, we call the functions that take the data as input and generate the emission map. The functions
+  ; to be called depend on the imaging algorithm.
+  
+  if method eq "EM" then begin
+    pixel_data_summed = stx_construct_pixel_data_summed(path_sci_file, time_range, energy_range, $
+                                                        path_bkg_file=path_bkg_file, xy_flare=xy_flare)
+
+    out_map = stx_em(pixel_data_summed, aux_data, imsize=imsize, pixel=pixel,mapcenter=mapcenter)
+
+  endif else begin
+    ; Compute calibrated visibilities
+    if keyword_set(no_small) then $
+      vis = stx_construct_calibrated_visibility(path_sci_file, time_range, energy_range, mapcenter, subc_index=subc_index, $
+      path_bkg_file=path_bkg_file, xy_flare=flare_loc, /no_small, sumcase='TOP+BOT') $
+    else vis = stx_construct_calibrated_visibility(path_sci_file, time_range, energy_range, mapcenter, subc_index=subc_index, $
+      path_bkg_file=path_bkg_file, xy_flare=flare_loc)
+
+    case method of
+      "MEM": out_map = stx_mem_ge(vis,imsize,pixel,aux_data,total_flux=max(abs(vis.obsvis)), /silent)
+      "CLEAN": begin
+        default, w_clean, 1  ; 1 = uniform weighting, 0 = natural weighting
+        niter  = 100   ; Number of iterations
+        gain   = 0.1   ; Gain used in each clean iteration
+        nmap   = 10    ; Plot clean components and cleaned map every 10 iterations
+        beam_width = 10. ; clean components are convolved with this beam
```

Comment:
I think that the user should have the possibility to set the CLEAN beam width value. What about adding a keyword 'clean_beam_width'? Also, if the considered detectors are 3-10, then a default value of the beam width of 10 arcsec is probably too small. Indeed, the resolution of the finest detectors (detectors 3a, 3b and 3c) is ~14 arcsec

Reply:
OK, done.